packages/opencensus-exporter-stackdriver/test/test-stackdriver-stats-utils.ts

```diff
@@ -413,4 +415,75 @@ describe('Stackdriver Stats Exporter Utils', () => {
          assert.deepStrictEqual(point2.value, {doubleValue: 15});
        });
   });
+
+  describe('getDefaultResource()', () => {
+    beforeEach(() => {
+      delete process.env.OC_RESOURCE_TYPE;
+      delete process.env.OC_RESOURCE_LABELS;
+      CoreResource.setup();
+    });
+
+    it('should return a global MonitoredResource', async () => {
+      nocks.noDetectResource();
+      const monitoredResource = await getDefaultResource('my-project-id');
+      const {type, labels} = monitoredResource;
+
+      assert.equal(type, 'global');
+      assert.deepEqual(labels, {project_id: 'my-project-id'});
+    });
+
+    it('should return a k8s MonitoredResource', async () => {
+      process.env.OC_RESOURCE_TYPE = 'k8s.io/container';
+      process.env.OC_RESOURCE_LABELS =
+          'k8s.io/pod/name=pod-xyz-123,k8s.io/container/name=c1,k8s.io/namespace/name=default,cloud.google.com/gce/zone=zone1';
```

Comment:
Optional nit: for long strings like this, I often like to break them up with the `+` operator to keep the line under 80 chars.